airflow/providers/docker/operators/docker_swarm.py

```diff
@@ -225,6 +231,24 @@ def stream_new_logs(last_line_logged, since=0):
             sleep(2)
             last_line_logged, last_timestamp = stream_new_logs(last_line_logged, since=last_timestamp)
 
+    @staticmethod
+    def format_args(args: list[str] | str | None) -> list[str] | None:
+        """Retrieve args.
+
+        The args string is parsed to a list. If it starts with ``[``,
+        the string is treated as a Python literal and parsed into a list.
+
+        :param args: args to the docker service
+
+        :return: the args as list
+        """
+        if isinstance(args, str):
+            if args.strip().startswith("["):
+                return ast.literal_eval(args)
+            else:
+                return shlex.split(args)
+        return args
```

Comment:
The parsing is necessary because docker-py expects the args as a list (https://docker-py.readthedocs.io/en/stable/api.html#docker.types.ContainerSpec). I added the string variants as a convenient option, that was not strictly needed.

Reply:
> I just got that from the docker command formatter

It was exists because :
- Field `DockerOperator.command` templated
- For historical reason, it add this support before NativeEnvironment and/or JSON encoder

But I can't see this case here

Reply:
I've replaced it with `json.loads`.

Reply:
I mean, maybe it is even not required, since it not templated field, so user have an option to provide it as pure string command

```python
DockerSwarmOperator(
    task_id="foo-bar",
    ...
    args="--foo bar --spam --egg",
)
```

Or command list

```python
DockerSwarmOperator(
    task_id="foo-bar",
    ...
    args=["--foo", "bar", "--spam", "--egg"],
)
```

But I'm not sure if it handy to provide as json / string representation of python list at all

```python
DockerSwarmOperator(
    task_id="foo-bar",
    ...
    args='["--foo", "bar", "--spam", "--egg"]',
)
```

Reply:
I see, yes, it's probably even better to leave that out.
Initially I thought to replicate the behavior of the docker command, as that is what the args are on the containerspec.
Thanks, I'll remove it.